subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/KotlinBuildScript.kt

```diff
@@ -44,6 +44,9 @@ import kotlin.script.templates.ScriptTemplateDefinition
         "-api-version", "1.8",
         "-Xjvm-default=all",
         "-Xjsr305=strict",
+        "-Xskip-metadata-version-check",
```

Comment:
It would be good to have something remind us to remove these once we are solidly transitioned to 2.0.

I mean, in general, for all these flag additions in various places. At least a TODO, if nothing else.

Reply:
We want to keep these flags long term. They allow flexibility in compiling scripts against different Kotlin versions and new Kotlin versions will keep happening.